enhancements/scorecard-storage.md

```diff
@@ -0,0 +1,333 @@
+    entrypoint:
+    - example-scorecard-test
+    - example-test
+    labels:
+      suite: custom
+      test: example-test
+    storage: 
+      spec:
+        storageClassName: hostpath
+        capacity:
+          storage: 1Gi
+        accessModes:
+          - ReadWriteOnce
+        mountPath:
+          path: /test-output
+```
+
+Valid values for access mode:
+ * ReadWriteOnce (default if not specified)
+ * ReadOnlyMany
+ * ReadWriteMany
+
+Access mode values are defined in https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/api/core/v1/types.go#L556
+
+Valid values for volume size must be formatted according to the rules defined for storage 
+in https://pkg.go.dev/k8s.io/apimachinery/pkg/api/resource#Format.  Typical values for 
+storage are like:
+ * 1Gi (1 gigabyte) (default)
+ * 100M (100 megabyte)
+
+Storage classes can be found on a cluster by the following command:
+ * kubectl get sc
+
+A default storage class is the storage class on a cluster that has the following annotation:
+```yaml
+  metadata:
+    annotations:
+      storageclass.kubernetes.io/is-default-class: "true"
+```
+
+
+#### Story 3
+Scorecard gathers any persisted volume data upon test completion and stores it in a local directory where scorecard is executed.  Users
+can specify a non-default local path for test output using a command flag as follows:
+```bash
+operator-sdk scorecard ./bundle --selector=suite=custom --test-output=/mytestoutput
+```
+
+#### Story 4
+A scorecard user can specify storage globally for any selected tests.
+The `storage` spec can be defined globally as in the following example:
+
+```yaml
+  storage: 
+    spec:
+      storageClassName: hostpath
+      capacity:
+        storage: 1Gi
+      accessModes:
+        - ReadWriteOnce
+      mountPath:
+        path: /test-output
+  tests:
+  - image: quay.io/custom-scorecard-example/example-test:latest
+    entrypoint:
+    - example-scorecard-test
+    - example-test
+    labels:
+      suite: example
+      test: example-test
+```
+
+This would mean that any selected test would assume the global storage
+settings.  If a test has its own unique test storage defined, then those
+settings would override the global storage settings.
+
+### Implementation Details/Notes/Constraints
+
+This feature, if specified by a scorecard user, requires a storage class be defined and operational on the k8s cluster nodes that scorecard tests would be executed upon.  If no storage class is specified or a default storage class is not available an error is returned for the scorecard test.
+
+This feature follows a workflow as follows:
+ * scorecard allocates a unique PVC for a test if that test requests storage
+ * scorecard mounts that PVC into the test Pod so the test can write to
+a given mountPath any test output they want
+ * upon test completion and before cleaning up the tests, scorecard
+will create a container that mounts the PVC solely to harvest (exec/tar)
```

Comment:
this seems to assume a single PV, but if each test has unique storage requirements then it'll be multiple PVs.

Reply:
(in fact below you do say it's a PV per test, so this workflow needs to iterate all the PVs, not just one)

Reply:
I'll make that clearer as it does interate across all tests/PVCs, not just one.

Reply:
updated.